Index: qca-ossl.cpp
===================================================================
--- qca-ossl.cpp	(revision 1300694)
+++ qca-ossl.cpp	(working copy)
@@ -1796,14 +1796,14 @@
 #endif
 		else if(alg == EMSA3_RIPEMD160)
 			md = EVP_ripemd160();
-		else if(alg == EMSA3_SHA224)
+/*		else if(alg == EMSA3_SHA224)
 			md = EVP_sha224();
 		else if(alg == EMSA3_SHA256)
 			md = EVP_sha256();
 		else if(alg == EMSA3_SHA384)
 			md = EVP_sha384();
 		else if(alg == EMSA3_SHA512)
-			md = EVP_sha512();
+			md = EVP_sha512();*/
 		else if(alg == EMSA3_Raw)
 		{
 			// md = 0
@@ -1824,14 +1824,14 @@
 #endif
 		else if(alg == EMSA3_RIPEMD160)
 			md = EVP_ripemd160();
-		else if(alg == EMSA3_SHA224)
+/*		else if(alg == EMSA3_SHA224)
 			md = EVP_sha224();
 		else if(alg == EMSA3_SHA256)
 			md = EVP_sha256();
 		else if(alg == EMSA3_SHA384)
 			md = EVP_sha384();
 		else if(alg == EMSA3_SHA512)
-			md = EVP_sha512();
+			md = EVP_sha512();*/
 		else if(alg == EMSA3_Raw)
 		{
 			// md = 0
@@ -3470,7 +3470,7 @@
 		case NID_dsaWithSHA1:
 		    p.sigalgo = QCA::EMSA1_SHA1;
 		    break;
-		case NID_sha224WithRSAEncryption:
+/*		case NID_sha224WithRSAEncryption:
 		    p.sigalgo = QCA::EMSA3_SHA224;
 		    break;
 		case NID_sha256WithRSAEncryption:
@@ -3481,7 +3481,7 @@
 		    break;
 		case NID_sha512WithRSAEncryption:
 		    p.sigalgo = QCA::EMSA3_SHA512;
-		    break;
+		    break;*/
 		default:
 		    qDebug() << "Unknown signature value: " << OBJ_obj2nid(x->cert_info->signature->algorithm);
 		    p.sigalgo = QCA::SignatureUnknown;
