diff -ru pkg-config-0.26.orig/pkg.h pkg-config-0.26/pkg.h
--- pkg-config-0.26.orig/pkg.h	2011-04-13 11:16:23 -0700
+++ pkg-config-0.26/pkg.h	2012-04-04 23:33:26 -0700
@@ -22,14 +22,22 @@
 
 #include <glib.h>
 
+static gchar *get_package_subdir(const gchar *subdir)
+{
+	gchar *p = g_win32_get_package_installation_directory_of_module(NULL);
+	gchar *p2 = g_build_filename(p, subdir, NULL);
+	g_free(p);
+	return p2;
+}
+
 #ifdef G_OS_WIN32
 /* No hardcoded paths in the binary, thanks */
 /* It's OK to leak this */
 #undef PKG_CONFIG_PC_PATH
 #define PKG_CONFIG_PC_PATH \
-  g_strconcat (g_win32_get_package_installation_subdirectory (NULL, NULL, "lib/pkgconfig"), \
+  g_strconcat (get_package_subdir ("lib/pkgconfig"), \
 	       ";", \
-	       g_win32_get_package_installation_subdirectory (NULL, NULL, "share/pkgconfig"), \
+	       get_package_subdir ("share/pkgconfig"), \
 	       NULL)
 #endif
 
